File: KI-304_Romaniuk_Lab5/src/main/java/KI/Romaniuk/Lab5/ResultFileHandler.java
package KI.Romaniuk.Lab5;

import java.io.*;

/**
 * Клас для читання і запису результатів у текстовому і двійковому форматах.
 */
public class ResultFileHandler {

    /**
     * Записує результат у текстовий файл.
     * @param result результат для запису
     * @param filePath шлях до текстового файлу
     * @throws IOException у випадку помилки запису
     */
    public void writeToTextFile(double result, String filePath) throws IOException {
        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
            writer.println("Результат: " + result);
        }
    }

    /**
     * Читає результат із текстового файлу.
     * @param filePath шлях до текстового файлу
     * @return зчитаний результат
     * @throws IOException у випадку помилки читання
     */
    public double readFromTextFile(String filePath) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
            String line = reader.readLine();
            if (line != null && line.startsWith("Результат: ")) {
                return Double.parseDouble(line.substring(10));
            } else {
                throw new IOException("Некоректний формат даних у файлі.");
            }
        }
    }

    /**
     * Записує результат у двійковий файл.
     * @param result результат для запису
     * @param filePath шлях до двійкового файлу
     * @throws IOException у випадку помилки запису
     */
    public void writeToBinaryFile(double result, String filePath) throws IOException {
        try (DataOutputStream dos = new DataOutputStream(new FileOutputStream(filePath))) {
            dos.writeDouble(result);
        }
    }

    /**
     * Читає результат із двійкового файлу.
     * @param filePath шлях до двійкового файлу
     * @return зчитаний результат
     * @throws IOException у випадку помилки читання
     */
    public double readFromBinaryFile(String filePath) throws IOException {
        try (DataInputStream dis = new DataInputStream(new FileInputStream(filePath))) {
            return dis.readDouble();
        }
    }
}

File: KI-304_Romaniuk_Lab8/result_file_handler.py
import struct

class ResultFileHandler:
    """
    Клас для роботи з текстовими та двійковими файлами.
    """

    def write_to_text_file(self, result, file_path):
        """
        Записує результат у текстовий файл.

        """
        try:
            with open(file_path, "w", encoding="utf-8") as file:
                file.write(f"Результат: {result}\n")
        except IOError as e:
            raise IOError(f"Помилка запису у файл {file_path}: {e}")

    def read_from_text_file(self, file_path):
        """
        Читає результат із текстового файлу.
        """
        try:
            with open(file_path, "r", encoding="utf-8") as file:
                line = file.readline()
                if line.startswith("Результат: "):
                    return float(line.split(": ")[1])
                else:
                    raise IOError("Некоректний формат даних у файлі.")
        except IOError as e:
            raise IOError(f"Помилка читання з файлу {file_path}: {e}")

    def write_to_binary_file(self, result, file_path):
        """
        Записує результат у двійковий файл.
        """
        try:
            with open(file_path, "wb") as file:
                file.write(struct.pack("d", result))  # "d" означає double
        except IOError as e:
            raise IOError(f"Помилка запису у файл {file_path}: {e}")

    def read_from_binary_file(self, file_path):
        """
        Читає результат із двійкового файлу.

        :param file_path: Шлях до файлу.
        :return: Зчитаний результат.
        :raises IOError: У випадку помилки читання.
        """
        try:
            with open(file_path, "rb") as file:
                return struct.unpack("d", file.read(8))[0]  # Читаємо double (8 байт)
        except IOError as e:
            raise IOError(f"Помилка читання з файлу {file_path}: {e}")
